src/clj_kondo/impl/docstring.clj

```diff
@@ -0,0 +1,53 @@
+(ns clj-kondo.impl.docstring
+  (:require [clj-kondo.impl.findings :as findings]
+            [clj-kondo.impl.utils :refer [node->line tag node->keyword string-from-token]]
+            [clojure.string :as str]))
+
+(defn docstring-messages
+  "Return a sequence of linting messages for the given `docstring`."
+  [docstring]
+  (cond-> nil
+    (str/blank? docstring)
+    (conj {:message "Docstring should not be blank."
+           :type :docstring-blank})
+
+    (not (re-find #"^\s*[A-Z].*[.!?]\s*$" docstring))
+    (conj {:message "First line of the docstring should be a capitalized sentence ending with punctuation."
+           :type :docstring-no-summary})
+
+    (or (re-find #"^\s" docstring)
+        (re-find #"\s$" docstring))
+    (conj {:message "Docstring should not have leading or trailing whitespace."
+           :type :docstring-leading-trailing-whitespace})))
+
+(defn lint-docstring!
+  "Lint `docstring` for styling issues.
+
+  `node` is the node reported when docstring has findings, so ideally
+  it should be the text node for `docstring`."
+  [{:keys [filename config] :as ctx} node docstring]
+  (when (some? docstring)
+    (doseq [{:keys [type message]} (docstring-messages docstring)]
+      (when-not (identical? :off (-> config :linters type :level))
```

Comment:
The purpose of doing these checks is to prevent unnecessary work if people aren't using these linters, but the work is done here anyway. This should change.